### PR TITLE
fix(discover): Fixed incorrect tags from showing in autocomplete

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
`count()` for example was showing up in the autocomplete if you had visited discover before visiting the performance landing page. The search bar would then incorrectly show it as a recommendation

This was caused by assigning into an existing object. The fix just assigning into a new object, but I switched`FIELD_TAGS` to a frozen / non-computed so that this can't happen again. I also switched field keys to an `Enum` so that it be used and also used with Record to make sure the field tags and values were correctly mapped.

#### Before
Screen Shot 2020-06-18 at 12 31 01 PM

#### After
Screen Shot 2020-06-18 at 12 30 06 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/discover/fields.tsx [hunk 1]
```diff
@@ -261,83 +261,385 @@ export type Aggregation = {
...
+  },
+  [FieldKey.PROJECT]: {
+    key: 'project',
+    name: 'project',
+  },
+  [FieldKey.ISSUE]: {
+    key: 'issue',
+    name: 'issue',
+  },
+}) as Readonly<Record<FieldKey, FieldTag>>;
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/discover/fields.tsx:639]: Is there not a way to generate this from the `FIELDS` and `FieldKey` definitions?

File: src/sentry/static/sentry/app/utils/discover/fields.tsx [hunk 2]
```diff
@@ -261,83 +261,385 @@ export type Aggregation = {
   multiPlotType: PlotType;
 };
 
+export enum FieldKey {
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/discover/fields.tsx:264]: Does this need to be exported?

File: src/sentry/static/sentry/app/utils/discover/fields.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/utils/discover/fields.tsx:639]: 👍 to generating this

File: src/sentry/static/sentry/app/utils/discover/fields.tsx [hunk 3]
```diff
@@ -261,83 +261,154 @@ export type Aggregation = {
...
+  [FieldKey.TRACE_PARENT_SPAN]: 'string',
 
   // Field alises defined in src/sentry/api/event_search.py
-  project: 'string',
-  issue: 'string',
+  [FieldKey.PROJECT]: 'string',
+  [FieldKey.ISSUE]: 'string',
 } as const;
-assert(FIELDS as Readonly<{[key in keyof typeof FIELDS]: ColumnType}>);
+assert(FIELDS as Readonly<Record<FieldKey, ColumnType>>);
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/discover/fields.tsx:399]: Is this assert() still necessary or could we define the type when defining the variable? like `const FIELDS: Readonly<Record<FieldKey, ColumnType>>  = {...}`

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/utils/discover/fields.tsx (hunk 3 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/utils/discover/fields.tsx:399]: I was considering it, but wasn't sure if leaving it with `as const` was preferable as it gives better type hinting?

Comment (k-fish) [reply src/sentry/static/sentry/app/utils/discover/fields.tsx:399]: nvm, checking shape will be a lot better for the mapping in the long wrong, thanks 👍

**Review by markstory — APPROVED** [review]

**Pull request merged**